casper/src/it/scala/coop.rchain.casper/HashSetCasperSpecification.scala

```diff
@@ -0,0 +1,212 @@
+import coop.rchain.casper.protocol.{BlockMessage, DeployData}
+import coop.rchain.casper.util.ProtoUtil
+import coop.rchain.casper.util.comm.TestNetwork
+import coop.rchain.crypto.codec.Base16
+import coop.rchain.crypto.hash.Keccak256
+import coop.rchain.crypto.signatures.{Ed25519, Secp256k1}
+import coop.rchain.rholang.interpreter.accounting
+import monix.execution.Scheduler.Implicits.global
+import coop.rchain.catscontrib._
+import coop.rchain.catscontrib.TaskContrib._
+import coop.rchain.comm.CommError
+import monix.eval.Task
+import org.scalacheck.{Gen, Prop, Properties, Shrink}
+import org.scalacheck.commands.Commands
+
+import scala.collection.immutable
+import scala.util.{Random, Try}
+
+case class RNode(idx: Int, name: String, deployed: Boolean)
+
+object HashSetCasperProperties extends Properties("HashSetCasper") {
+  implicit def noShrink[T]: Shrink[T] = Shrink.shrinkAny
+  property("network") = HashSetCasperSpecification.property()
+}
+
+
+class NodeBox(val node: HashSetCasperTestNode[Effect], var lastBlock: Option[BlockMessage]) {
+  def update(bm: Option[BlockMessage]): Unit = this.lastBlock = bm
+}
+
+object HashSetCasperActions {
+  import HashSetCasperTest._
+
+  def context(amount: Int, bondsGen: Seq[Array[Byte]] => Map[Array[Byte], Long])
+    : (BlockMessage, immutable.IndexedSeq[Array[Byte]]) = {
+    val (validatorKeys, validators) = (1 to amount).map(_ => Ed25519.newKeyPair).unzip
+    val (_, ethPubKeys)    = (1 to amount).map(_ => Secp256k1.newKeyPair).unzip
+    val ethAddresses =
+      ethPubKeys.map(pk => "0x" + Base16.encode(Keccak256.hash(pk.bytes.drop(1)).takeRight(20)))
+    val wallets = ethAddresses.map(addr => PreWallet(addr, BigInt(10001)))
+    val bonds = bondsGen(validators)
+    val minimumBond = 100L
+    val genesis =
+      buildGenesis(wallets, bonds, minimumBond, Long.MaxValue, Faucet.basicWalletFaucet, 0L)
+    (genesis, validatorKeys)
+  }
+
+  def deploy(
+              node: HashSetCasperTestNode[Effect],
+              dd: DeployData
+            ): Effect[Either[Throwable, Unit]] = node.casperEff.deploy(dd)
+
+  def create(
+              node: HashSetCasperTestNode[Effect]
+            ): EitherT[Task, CommError, BlockMessage] =
+    for {
+      createBlockResult1    <- node.casperEff.createBlock
+      Created(signedBlock1) = createBlockResult1
+    } yield signedBlock1
+
+  def add(node: HashSetCasperTestNode[Effect], signed: BlockMessage): Effect[Either[Throwable, BlockStatus]] =
+    Sync[Effect].attempt(
+      node.casperEff.addBlock(signed, ignoreDoppelgangerCheck[Effect])
+    )
+
+  def deployment(i: Int, ts: Long = System.currentTimeMillis()): DeployData =
+    ProtoUtil.sourceDeploy(s"new x in { x!(0) }", ts, accounting.MAX_VALUE)
+}
+
+object HashSetCasperSpecification extends Commands {
+  implicit def noShrink[T]: Shrink[T] = Shrink.shrinkAny
+  import HashSetCasperActions._
+
+  type State = List[RNode]
+  type Sut = List[NodeBox]
```

Comment:
Could you add `override`s where applicable? I guess both type `State` and `Sut` come from a superclass